client/src/types/data.ts

```diff
@@ -0,0 +1,109 @@
+
+/** Value runtype of humidity sensor data */
+export const HumidityRT = Number;
+
+/** Value runtype of steeringAngle sensor data */
+export const SteeringAngleRT = Number;
+
+/** Value runtype of co2 sensor data */
+export const CO2RT = Number;
+
+/** Value runtype of accelerometer sensor data */
+export const AccelerometerRT = Record({
+    /** X value */
+    x: Number,
+    /** Y value */
+    y: Number,
+    /** Z value */
+    z: Number
+});
+
+/** Value runtype of gyroscope sensor data */
+export const GyroscopeRT = Record({
+    /** X value */
+    x: Number,
+    /** Y value */
+    y: Number,
+    /** Z value */
+    z: Number
+});
+
+/** Value runtype of reedVelocity sensor data */
+export const ReedVelocityRT = Number;
+
+/** Value runtype of reedDistance sensor data */
+export const ReedDistanceRT = Number;
+
+/** Value runtype of gps sensor data */
+export const GPSRT = Record({
+    /** Speed */
+    speed: Number,
+    // /** Dilution of precision */
+    // pdop: Number,
+    /** Latitude */
+    latitude: Number,
+    /** Longitude */
+    longitude: Number,
+    /** Altitude */
+    altitude: Number,
+    /** Course */
+    course: Number,
+    // /** Datetime */
+    // datetime: String
+});
+
+/** Value runtype of power sensor data */
+export const PowerRT = Number;
+
+/** Value runtype of cadence sensor data */
+export const CadenceRT = Number;
+
+/** Value runtype of heartRate sensor data */
+export const HeartRateRT = Number;
+
+/* ------------------------------------ Types ------------------------------------ */
+
+/** Value type of temperature sensor data */
+export type TemperatureT = Static<typeof TemperatureRT>
+
+/** Value type of humidity sensor data */
+export type HumidityT = Static<typeof HumidityRT>
+
+/** Value type of steeringAngle sensor data */
+export type SteeringAngleT = Static<typeof SteeringAngleRT>
+
+/** Value type of co2 sensor data */
+export type CO2T = Static<typeof CO2RT>
+
+/** Value type of accelerometer sensor data */
+export type AccelerometerT = Static<typeof AccelerometerRT>
+
+/** Value type of gyroscope sensor data */
+export type GyroscopeT = Static<typeof GyroscopeRT>
+
+/** Value type of reedVelocity sensor data */
+export type ReedVelocityT = Static<typeof ReedVelocityRT>
+
+/** Value type of reedDistance sensor data */
+export type ReedDistanceT = Static<typeof ReedDistanceRT>
+
+/** Value type of gps sensor data */
+export type GPST = Static<typeof GPSRT>
+
+/** Value type of power sensor data */
+export type PowerT = Static<typeof PowerRT>
+
+/** Value type of cadence sensor data */
+export type CadenceT = Static<typeof CadenceRT>
+
+/** Value type of heartRate sensor data */
+export type HeartRateT = Static<typeof HeartRateRT>
+
+/** Union type of all sensor value types */
+export type SensorsT = TemperatureT | HumidityT | SteeringAngleT | CO2T | AccelerometerT | GyroscopeT | ReedVelocityT | ReedDistanceT | GPST | PowerT | CadenceT | HeartRateT;
```

Comment:
Run `prettier` on your modified files, will convert to 2 space indents and make long lines like this more managable.

I'm thinking I might set up a `prettier` pre-commit hook on the dashboard repo, meaning the code is automatically formatted before a commit is made. Thoughts on this? @kdav108 @Blake-Haydon also

Reply:
I think it's a good idea, though I thought eslint already extended prettier and deals with that sort of formatting stuff? My formatter is set to follow the eslint formatting rules but if we want to use specifically use prettier that's fine too, will just have to change the project configs.

Reply:
1782b29

Reply:
Hmm didn't realise prettier wasn't part of the project, I must have gotten confused with what we use at work. I reckon it's pretty neat but don't worry about it for now then.